File: src/main/java/land/leets/domain/application/usecase/GetApplicationDetails.java
package land.leets.domain.application.usecase;

import land.leets.domain.application.domain.Application;
import land.leets.domain.application.presentation.dto.ApplicationDetailsResponse;

import java.util.UUID;

public interface GetApplicationDetails {
    ApplicationDetailsResponse execute(Long id);
    Application execute(UUID uid);
}

File: src/main/kotlin/land/leets/domain/application/usecase/GetApplicationDetails.kt
package land.leets.domain.application.usecase

import land.leets.domain.application.domain.Application
import land.leets.domain.application.presentation.dto.ApplicationDetailsResponse
import java.util.UUID

interface GetApplicationDetails {
    fun execute(id: Long): ApplicationDetailsResponse
    fun execute(uid: UUID): Application
}
